### ISSUE TITLE
How to debug error: Connecting to test.docksal (test.docksal)|192.168.64.100|:80... failed: Connection refused.

### ISSUE THREAD

**Issue by jessehs** [docksal/docksal#939]
**Description**

I have an issue which I believe is a system-wide config problem. I have not seen this with the "common problems" troubleshooting section, and I'm at a loss as to the next debugging steps.

By posting this issue I'm hoping to get some more suggestions on the next debugging steps. I believe I have a problem with my system where I always get the same error:

`Connecting to test.docksal (test.docksal)|192.168.64.100|:80... failed: Connection refused.`

**Steps to reproduce the issue and debugging steps performed:**

I installed Docksal with Docker 4 Mac.

Some steps I've tried:

Performing "fin system stop; fin system start"
Restarting the host computer (macbook OS X v. 10.14.1)
Resetting Docker for mac to factory defaults.

I've tried creating a few different project and get the same error. To break it down to the most simple case, I've created a very simple directory "test" with 4 files:

test/.docksal/docksal.env (DOCKSAL_STACK=default / DOCROOT=docroot)
test/.docksal/docksal.yml (version 2.1)
test/docroot/index.php (phpinfo();)
test/docroot/index.html (Hello world)

In my ~/.docksal/docksal.env I have:

DOCKER_NATIVE=1
DOCKSAL_UUID=f0788bf1-1ef3-4aef-33c8-f2bfa7c80bb3
DOCKSAL_DNS_DEBUG=true

```
$ fin system start
Starting Docksal system services...
Configuring network settings...
Password:
Resetting Docksal services...
 ➔ proxy
 ➔ dns
   upstream 8.8.8.8
 ➔ ssh-agent
Identity added: id_rsa (id_rsa)
```

```
$ fin start
Key 'id_rsa' already loaded in the agent. Skipping.
Starting services...
Creating network "test_default" with the default driver
Starting test_cli_1 ... done
Starting test_db_1  ... done
Starting test_web_1 ... done
Waiting 10s for container to start...
Connected vhost-proxy to "test_default" network.
Virtual Host: http://test.docksal
```

```
$ wget test.docksal
--2019-01-23 17:17:11--  http://test.docksal/
Resolving test.docksal... 192.168.64.100
Connecting to test.docksal|192.168.64.100|:80... failed: Connection refused.
```

In the log for the dns container:

```
$ docker logs -f docksal-dns
2019-01-24 00:14:52 | Generating configuration in /etc/dnsmasq.d/docksal.conf
2019-01-24 00:14:52 | Passing execution to: dnsmasq
dnsmasq: started, version 2.79 cachesize 150
dnsmasq: compile time options: IPv6 GNU-getopt no-DBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth no-DNSSEC loop-detect inotify
dnsmasq: reading /etc/resolv.conf
dnsmasq: using nameserver 8.8.8.8#53
dnsmasq: using nameserver 9.9.9.9#53
dnsmasq: read /etc/hosts - 7 addresses

dnsmasq: query[A] test.docksal from 172.17.0.1
dnsmasq: config test.docksal is 192.168.64.100
```

Output for "docker ps":

```
$ docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                   PORTS                                                    NAMES
3e04f80f7bb5        docksal/ssh-agent:1.2       "docker-entrypoint.s…"   4 minutes ago       Up 4 minutes (healthy)                                                            docksal-ssh-agent
7125bce24b63        docksal/dns:1.1             "docker-entrypoint.s…"   4 minutes ago       Up 4 minutes (healthy)   192.168.64.100:53->53/udp                                docksal-dns
c7d3aa490f28        docksal/vhost-proxy:1.4     "docker-entrypoint.s…"   4 minutes ago       Up 4 minutes (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
79c405329799        docksal/web:2.1-apache2.4   "httpd-foreground"       17 minutes ago      Up 3 minutes             80/tcp, 443/tcp                                          test_web_1
4f8f2fd567d7        docksal/cli:2.5-php7.1      "/opt/startup.sh sup…"   17 minutes ago      Up 3 minutes (healthy)   22/tcp, 3000/tcp, 9000/tcp                               test_cli_1
8932de182cb2        docksal/db:1.2-mysql-5.6    "/entrypoint.sh mysq…"   17 minutes ago      Up 3 minutes             0.0.0.0:32770->3306/tcp                                  test_db_1

```

The command "fin docker inspect test_web_1" reveals "IPAddress": "172.21.0.4".

Then I try accessing the test file from within the "cli" container:

```
$ fin bash cli

$ wget -O - 172.21.0.4/index.html

```
I see the "Hello World" output from my index.html file. Same thing goes for the index.php file. I see the output of the phpinfo command. There is no DNS logging reported during these 2 tests.

Then I also try from inside the cli container:

```
$ wget -O - test.docksal/index.php
--2019-01-24 00:27:04--  http://test.docksal/index.php
Resolving test.docksal (test.docksal)... 192.168.64.100
Connecting to test.docksal (test.docksal)|192.168.64.100|:80... failed: Connection refused.

```
This is associated with the DNS logs:

```
dnsmasq: query[AAAA] test.docksal from 172.17.0.1
dnsmasq: config test.docksal is NODATA-IPv6
dnsmasq: query[A] test.docksal from 172.17.0.1
dnsmasq: config test.docksal is 192.168.64.100
```

**Output of `fin config`:**

<details>
  <summary>fin config output</summary>

```
---------------------
COMPOSE_PROJECT_NAME_SAFE: test
COMPOSE_FILE:
/Users/jhofmann-smith/.docksal/stacks/overrides-osxfs.yml
/Users/jhofmann-smith/.docksal/stacks/volumes-bind.yml
/Users/jhofmann-smith/.docksal/stacks/stack-default.yml
/Users/jhofmann-smith/projects/test/.docksal/docksal.yml
ENV_FILE:
/Users/jhofmann-smith/projects/test/.docksal/docksal.env

PROJECT_ROOT: /Users/jhofmann-smith/projects/test
DOCROOT: docroot
VIRTUAL_HOST: test.docksal
VIRTUAL_HOST_ALIASES: *.test.docksal
IP: 192.168.64.100
MYSQL: 192.168.64.100:32770

Docker Compose configuration
---------------------
services:
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
      DOCROOT: docroot
      GIT_USER_EMAIL: jhofmann-smith@phase2technology.com
      GIT_USER_NAME: Jesse Hofmann-Smith
      HOST_GID: '20'
      HOST_UID: '502'
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: test.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: docksal/cli:2.5-php7.1
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - cli_home:/home/docker:rw
    - /Users/jhofmann-smith/projects/test:/var/www:rw,cached
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_ALLOW_EMPTY_PASSWORD: null
      MYSQL_DATABASE: default
      MYSQL_INITDB_SKIP_TZINFO: null
      MYSQL_ONETIME_PASSWORD: null
      MYSQL_PASSWORD: user
      MYSQL_RANDOM_ROOT_PASSWORD: null
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.2-mysql-5.6
    ports:
    - 3306/tcp
    volumes:
    - db_data:/var/lib/mysql:rw
    - project_root:/var/www:ro,nocopy
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/docroot
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.cert-name: none
      io.docksal.permanent: "false"
      io.docksal.project-root: /Users/jhofmann-smith/projects/test
      io.docksal.virtual-host: test.docksal,*.test.docksal,test.docksal.*
    volumes:
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  cli_home: {}
  db_data: {}
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Users/jhofmann-smith/projects/test
      o: bind
      type: none

---------------------
```

</details>

<br>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
███  OS
Darwin Mac OS X 10.14.1
Darwin Jesses-MacBook-Pro.local 18.2.0 Darwin Kernel Version 18.2.0: Fri Oct  5 19:41:49 PDT 2018; root:xnu-4903.221.2~2/RELEASE_X86_64 x86_64

███  ENVIRONMENT
MODE : Docker for Mac

███  FIN
fin version: 1.80.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.23.1
docker-compose version 1.23.2, build 1110ad01
docker-py version: 3.6.0
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.1.0h  27 Mar 2018

███  DOCKER
EXPECTED VERSION: 18.06.1-ce

Client: Docker Engine - Community
 Version:           18.09.1
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        4c52b90
 Built:             Wed Jan  9 19:33:12 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.1
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       4c52b90
  Built:            Wed Jan  9 19:41:49 2019
  OS/Arch:          linux/amd64
  Experimental:     false

███  DOCKSAL: PROJECTS
project             STATUS              virtual host                                 project root
test                Up 27 minutes       test.docksal,*.test.docksal,test.docksal.*   /Users/jhofmann-smith/projects/test

███  DOCKSAL: VIRTUAL HOSTS
*.test.docksal
test.docksal.*
test.docksal

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS                    PORTS                                                    NAMES
3e04f80f7bb5        docksal/ssh-agent:1.2       "docker-entrypoint.s…"   27 minutes ago      Up 27 minutes (healthy)                                                            docksal-ssh-agent
7125bce24b63        docksal/dns:1.1             "docker-entrypoint.s…"   27 minutes ago      Up 27 minutes (healthy)   192.168.64.100:53->53/udp                                docksal-dns
c7d3aa490f28        docksal/vhost-proxy:1.4     "docker-entrypoint.s…"   27 minutes ago      Up 27 minutes (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
79c405329799        docksal/web:2.1-apache2.4   "httpd-foreground"       40 minutes ago      Up 27 minutes             80/tcp, 443/tcp                                          test_web_1
4f8f2fd567d7        docksal/cli:2.5-php7.1      "/opt/startup.sh sup…"   40 minutes ago      Up 27 minutes (healthy)   22/tcp, 3000/tcp, 9000/tcp                               test_cli_1
8932de182cb2        docksal/db:1.2-mysql-5.6    "/entrypoint.sh mysq…"   40 minutes ago      Up 27 minutes             0.0.0.0:32770->3306/tcp                                  test_db_1

███  DOCKER: NETWORKS
NETWORK ID          NAME                DRIVER              SCOPE
87d4c0a2c663        _default            bridge              local
9f1bc838e705        bridge              bridge              local
04368a091cad        host                host                local
97cb73f99ffc        none                null                local
cdd269845c12        test_default        bridge              local

███  HDD Usage
Filesystem      Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1   466Gi  398Gi   66Gi    86% 3591776 9223372036851184031    0%   /
devfs          352Ki  352Ki    0Bi   100%    1218                   0  100%   /dev
/dev/disk1s4   466Gi  1.0Gi   66Gi     2%       1 9223372036854775806    0%   /private/var/vm
map -hosts       0Bi    0Bi    0Bi   100%       0                   0  100%   /net
map auto_home    0Bi    0Bi    0Bi   100%       0                   0  100%   /home
/dev/disk1s3   466Gi  489Mi   66Gi     1%      20 9223372036854775787    0%   /Volumes/Recovery  ```

</details>

**Comment by lmakarov** [comment]
@jessehs do you have firewall/antivirus/vpn enabled? If so, try disabling one by one and test.

Also, check `vhost-proxy` logs - `fin docker logs docksal-vhost-proxy`.

**Comment by jessehs** [comment]
Thanks for your help @lmakarov! I don't believe that I have any firewall, antivirus or VPN running.

I don't know much about routing tables, etc, but maybe you can make sense of this. I launched the Network Utility app and ran Netstat for routing table info both before and after launching the "fin system start" command. After performing 'fin system start' I see the entries for "192.168.64.1" and "192.168.64.100". Also, before starting, the ping command to 192.168.64.100 does not return a response, but after starting it does.

Before:

```
Routing tables

Internet:
Destination        Gateway            Flags        Refs      Use   Netif Expire
default            192.168.1.1        UGSc          268        0     en0       
127                localhost          UCS             0        0     lo0       
localhost          localhost          UH              8      352     lo0       
169.254            link#5             UCS             0        0     en0      !
192.168.1          link#5             UCS             0        0     en0      !
192.168.1.1/32     link#5             UCS             1        0     en0      !
192.168.1.1        8c:3b:ad:41:54:99  UHLWIir       230       16     en0   1192
192.168.1.24/32    link#5             UCS             0        0     en0      !
224.0.0/4          link#5             UmCS            2        0     en0      !
224.0.0.251        1:0:5e:0:0:fb      UHmLWI          0        0     en0       
239.255.255.250    1:0:5e:7f:ff:fa    UHmLWI          0       33     en0       
255.255.255.255/32 link#5             UCS             0        0     en0      !

Internet6:
Destination        Gateway            Flags         Netif Expire
default            fe80::%utun0       UGcI          utun0       
localhost          localhost          UHL             lo0       
fe80::%lo0         localhost          UcI             lo0       
localhost          link#1             UHLI            lo0       
fe80::%en0         link#5             UCI             en0       
jesses-macbook-pro 6c:96:cf:dc:95:97  UHLI            lo0       
fe80::%awdl0       link#7             UCI           awdl0       
fe80::40c8:2eff:fe 42:c8:2e:cb:5f:7f  UHLI            lo0       
fe80::%utun0       jesses-macbook-pro UcI           utun0       
jesses-macbook-pro link#11            UHLI            lo0       
ff01::%lo0         localhost          UmCI            lo0       
ff01::%en0         link#5             UmCI            en0       
ff01::%awdl0       link#7             UmCI          awdl0       
ff01::%utun0       jesses-macbook-pro UmCI          utun0       
ff02::%lo0         localhost          UmCI            lo0       
ff02::%en0         link#5             UmCI            en0       
ff02::%awdl0       link#7             UmCI          awdl0       
ff02::%utun0       jesses-macbook-pro UmCI          utun0   
```

After:

```
Routing tables

Internet:
Destination        Gateway            Flags        Refs      Use   Netif Expire
default            192.168.1.1        UGSc          276        0     en0       
127                localhost          UCS             0        0     lo0       
localhost          localhost          UH              8      336     lo0       
169.254            link#5             UCS             0        0     en0      !
192.168.1          link#5             UCS             0        0     en0      !
192.168.1.1/32     link#5             UCS             1        0     en0      !
192.168.1.1        8c:3b:ad:41:54:99  UHLWIir       224       16     en0   1191
192.168.1.24/32    link#5             UCS             0        0     en0      !
192.168.64.1       192.168.64.1       UH              0        0     lo0       
192.168.64.100     192.168.64.100     UH              1        4     lo0       
224.0.0/4          link#5             UmCS            2        0     en0      !
224.0.0.251        1:0:5e:0:0:fb      UHmLWI          0        0     en0       
239.255.255.250    1:0:5e:7f:ff:fa    UHmLWI          0       28     en0       
255.255.255.255/32 link#5             UCS             0        0     en0      !

Internet6:
Destination        Gateway            Flags         Netif Expire
default            fe80::%utun0       UGcI          utun0       
localhost          localhost          UHL             lo0       
fe80::%lo0         localhost          UcI             lo0       
localhost          link#1             UHLI            lo0       
fe80::%en0         link#5             UCI             en0       
jesses-macbook-pro 6c:96:cf:dc:95:97  UHLI            lo0       
fe80::%awdl0       link#7             UCI           awdl0       
fe80::40c8:2eff:fe 42:c8:2e:cb:5f:7f  UHLI            lo0       
fe80::%utun0       jesses-macbook-pro UcI           utun0       
jesses-macbook-pro link#11            UHLI            lo0       
ff01::%lo0         localhost          UmCI            lo0       
ff01::%en0         link#5             UmCI            en0       
ff01::%awdl0       link#7             UmCI          awdl0       
ff01::%utun0       jesses-macbook-pro UmCI          utun0       
ff02::%lo0         localhost          UmCI            lo0       
ff02::%en0         link#5             UmCI            en0       
ff02::%awdl0       link#7             UmCI          awdl0       
ff02::%utun0       jesses-macbook-pro UmCI          utun0       
```
Performing the "traceroute" command before and after fin system start, I get this.

Before:

```
traceroute to 192.168.64.100 (192.168.64.100), 64 hops max, 72 byte packets
 1  192.168.1.1 (192.168.1.1)  1.431 ms  0.944 ms  2.077 ms
 2  * * *
```

After:

```
traceroute to 192.168.64.100 (192.168.64.100), 64 hops max, 72 byte packets
 1  192.168.64.100 (192.168.64.100)  0.358 ms  0.121 ms  0.109 ms
```

After the fin system start, running the fin docker logs docksal-vhost-proxy command, and then running "fin start" in the test directory, I see this (comments added):

```
fin docker logs -f docksal-vhost-proxy


######### After 'fin system start'


2019-01-24 19:49:51 [proxyctl]  Connected proxy to network: _default.
2019-01-24 19:49:51,778 INFO Set uid to user 0 succeeded
2019-01-24 19:49:51,786 INFO RPC interface 'supervisor' initialized
2019-01-24 19:49:51,787 INFO supervisord started with pid 1
2019-01-24 19:49:52,789 INFO spawned: 'nginx' with pid 40
2019-01-24 19:49:52,791 INFO spawned: 'crond' with pid 41
2019-01-24 19:49:52,793 INFO spawned: 'docker-gen' with pid 42
crond: crond (busybox 1.28.4) started, log level 8
2019/01/24 19:49:53 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
2019/01/24 19:49:53 Watching docker events
2019/01/24 19:49:53 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
2019-01-24 19:49:54,079 INFO success: nginx entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019/01/24 19:49:54 Received event start for container 8ac655c59cc9
2019-01-24 19:49:54,079 INFO success: crond entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019-01-24 19:49:54,079 INFO success: docker-gen entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2019/01/24 19:49:54 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
crond: USER root pid  64 cmd /usr/local/bin/proxyctl cron
2019-01-24 19:50:00 [proxyctl] [cron]
2019-01-24 19:50:00 [proxyctl] [stop]


######### After 'fin start' in test directory


2019/01/24 19:50:40 Received event start for container 8932de182cb2
2019/01/24 19:50:40 Received event start for container 4f8f2fd567d7
2019/01/24 19:50:41 Received event start for container 79c405329799
2019/01/24 19:50:41 Generated '/etc/nginx/conf.d/vhosts.conf' from 6 containers
2019/01/24 19:50:41 Running 'proxyctl notify'
2019/01/24 19:50:41 Error running notify command: proxyctl notify, exit status 1
2019/01/24 19:50:41 [proxyctl notify]: 2019/01/24 19:50:41 [emerg] 172#172: no servers are inside upstream in /etc/nginx/conf.d/vhosts.conf:4
2019/01/24 19:50:41 [proxyctl notify]: nginx: [emerg] no servers are inside upstream in /etc/nginx/conf.d/vhosts.conf:4
2019/01/24 19:50:41 [proxyctl notify]: nginx: configuration file /usr/local/openresty/nginx/conf/nginx.conf test failed
2019/01/24 19:50:41 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
2019/01/24 19:50:41 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
2019/01/24 19:50:52 Received event start for container 5e05a5b3a129
2019/01/24 19:50:52 Generated '/etc/nginx/conf.d/vhosts.conf' from 7 containers
2019/01/24 19:50:52 Running 'proxyctl notify'
2019/01/24 19:50:52 [proxyctl notify]: nginx: the configuration file /usr/local/openresty/nginx/conf/nginx.conf syntax is ok
2019/01/24 19:50:52 [proxyctl notify]: nginx: configuration file /usr/local/openresty/nginx/conf/nginx.conf test is successful
2019/01/24 19:50:52 [proxyctl notify]: 2019/01/24 19:50:52 [notice] 189#189: signal process started
2019/01/24 19:50:52 Received event die for container 5e05a5b3a129
2019/01/24 19:50:53 Contents of /etc/nginx/conf.d/vhosts.conf did not change. Skipping notification 'proxyctl notify'
crond: USER root pid 204 cmd /usr/local/bin/proxyctl cron
2019-01-24 19:51:00 [proxyctl] [cron]
2019-01-24 19:51:00 [proxyctl] [stop]
```

Then, I tried looking inside the vhost-proxy container for the nginx config:

####### enter vhost proxy container:

```
$ fin bash docksal-vhost-proxy

$ vi /etc/nginx/conf.d/vhosts.conf


        # -------------------------------------------------- #
        # Reachable via "test_default" network
        upstream test-web {
                # test_web_1
                server 172.19.0.4:80;
        }
        ## HTTPS
        server {
                listen 80;
                listen 443 ssl http2;
                server_name test.docksal;
                server_name *.test.docksal;
                server_name test.docksal.*;
                ssl_certificate /etc/certs/server.crt;
                ssl_certificate_key /etc/certs/server.key;
                ssl_session_cache builtin:1000 shared:SSL:10m;
                ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
                ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
                ssl_prefer_server_ciphers on;
                location / {
                        proxy_pass http://test-web;
                }
        }
        # -------------------------------------------------- #
```

**Comment by lmakarov** [comment]
@jessehs so you are able to `ping 192.168.64.100` after starting Docksal, but getting this from wget?

> ```
> $ wget test.docksal
> --2019-01-23 17:17:11--  http://test.docksal/
> Resolving test.docksal... 192.168.64.100
> Connecting to test.docksal|192.168.64.100|:80... failed: Connection refused.
> ```
> 

It looks like everything is up and running and vhost-proxy is healthy. Not sure what else can be blocking `192.168.64.100:80`, other than a firewall/antivirus/vpn.

How about `wget https://test.docksal`?

Try stopping all projects (`fin stop --all`), doing a system reset (`fin system reset`) and checking `wget http://test.docksal` again.

**Comment by achekulaev** [comment]
```
# Set Docksal HTTP port to 8080
fin config set --global DOCKSAL_VHOST_PROXY_PORT_HTTP=8080
# Apply the change
fin system reset
# Test if it works now
wget http://test.docksal:8080
```

If above works you definitely have some other software blocking the connection.

**Comment by achekulaev** [comment]
> Using the Network Utility "port scan" feature, after the "fin system reset" command, I get the following:

OK. So looks like we're onto something here. You definitely should get ports 80 and 443 there. Can you scan it now and show the output? Does it have 8080? (I suppose no)

**Comment by achekulaev** [comment]
@jessehs show output of `ifconfig`

**Comment by jessehs** [comment]
```
ifconfig
lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet 127.0.0.1 netmask 0xff000000
	inet6 ::1 prefixlen 128
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1
	inet 192.168.64.100 netmask 0xffffff00
	inet 192.168.64.1 netmask 0xffffff00
	nd6 options=201<PERFORMNUD,DAD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
XHC20: flags=0<> mtu 0
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether 6c:96:cf:dc:95:97
	inet6 fe80::1066:4a4:c8a8:730b%en0 prefixlen 64 secured scopeid 0x5
	inet 192.168.1.24 netmask 0xffffff00 broadcast 192.168.1.255
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
p2p0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 2304
	ether 0e:96:cf:dc:95:97
	media: autoselect
	status: inactive
awdl0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1484
	ether ea:81:49:5b:55:1b
	inet6 fe80::e881:49ff:fe5b:551b%awdl0 prefixlen 64 scopeid 0x7
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
en1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=60<TSO4,TSO6>
	ether 6a:00:03:68:fc:b0
	media: autoselect <full-duplex>
	status: inactive
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=60<TSO4,TSO6>
	ether 6a:00:03:68:fc:b1
	media: autoselect <full-duplex>
	status: inactive
bridge0: flags=8822<BROADCAST,SMART,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether 6a:00:03:68:fc:b0
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x2
	member: en1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 8 priority 0 path cost 0
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 9 priority 0 path cost 0
	media: <unknown type>
	status: inactive
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::e7aa:8199:7abf:30c8%utun0 prefixlen 64 scopeid 0xb
	nd6 options=201<PERFORMNUD,DAD>
```

**Comment by jessehs** [comment]
Also, yes,  no port 8080:

```
Port Scanning host: 192.168.64.100

	 Open TCP Port: 	111    		sunrpc
	 Open TCP Port: 	999    		garcon
	 Open TCP Port: 	1017
	 Open TCP Port: 	1021   		exp1
	 Open TCP Port: 	1023
	 Open TCP Port: 	2049   		nfsd

```

**Comment by lmakarov** [comment]
@jessehs try launching a simple nginx container on port `8000`:

```
fin docker run --rm -p 8000:80 nginx
```

With Docker for Mac you should be able to see the nginx welcome page on both:

```
http://127.0.0.1:8000
http://192.168.64.100:8000
```

If that works, try:

```
fin config set --global DOCKSAL_VHOST_PROXY_IP=0.0.0.0
fin system reset vhost-proxy
```

That will expose vhost-proxy on all interfaces, including localhost (127.0.0.1):

```
http://127.0.0.1
```

(or `http://127.0.0.1:8080` if you still have `DOCKSAL_VHOST_PROXY_PORT_HTTP=8080` configured)

**Comment by jessehs** [comment]
Success! Thank you both so much. When I have _both_ of these set in my ~/.docksal/docksal.env file and I perform `fin system reset` then the http://test.docksal domain works. I can see the output of phpinfo() in the index.php file. I really appreciate the support resolving this!

```
DOCKSAL_VHOST_PROXY_PORT_HTTP="8080"
DOCKSAL_VHOST_PROXY_IP="0.0.0.0"
```

It does not work when either one of them is disabled. Docker ps now looks like this:

```
docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED              STATUS                        PORTS                                        NAMES
ba5891b95e65        docksal/ssh-agent:1.2       "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)                                                docksal-ssh-agent
16f39262fe7e        docksal/dns:1.1             "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   192.168.64.100:53->53/udp                    docksal-dns
dfb0e022c47d        docksal/vhost-proxy:1.4     "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   0.0.0.0:443->443/tcp, 0.0.0.0:8080->80/tcp   docksal-vhost-proxy
766b04b72d70        docksal/web:2.1-apache2.4   "httpd-foreground"       17 hours ago         Up About a minute             80/tcp, 443/tcp                              test_web_1
cd9b86c21e03        docksal/cli:2.5-php7.1      "/opt/startup.sh sup…"   17 hours ago         Up About a minute (healthy)   22/tcp, 3000/tcp, 9000/tcp                   test_cli_1
1bbe94a18dce        docksal/db:1.2-mysql-5.6    "/entrypoint.sh mysq…"   17 hours ago         Up About a minute             0.0.0.0:32771->3306/tcp                      test_db_1
```

**Comment by lmakarov** [comment]
@jessehs In the output above `vhost-proxy` listens on `0.0.0.0:8080`, so it would not respond on http://test.docksal, but rather http://test.docksal:8080. Can you please confirm this?

Also, can you confirm which IP address `test.docksal` (or `<anything>.docksal` for that matter) resolves to for you?

**Comment by jessehs** [comment]
So it looks like it responds successfully the same way to both http://test.dockal and http://test.docksal:8080.

The IP that responds to a ping of test.docksal is 192.168.64.100.

**Comment by lmakarov** [comment]
@jessehs I'm glad your issues is resolved, however I don't wont it to confuse others landing here.

based on your `docker ps` output in https://github.com/docksal/docksal/issues/939#issuecomment-457613438, no container has port 80 exposed. `vhost-proxy` binds strictly to `8080`.

Are you sure you were not mixing `http://` (port 80 or 8080 as you have it configured above) vs `https://` (port 443)? There must be a discrepancy in the results you shared, as they do not make much sense.

**Comment by lmakarov** [comment]
@jessehs do you by any chance have Docksal running with both VirtualBox AND Docker Desktop at the same time?

**Comment by vflirt** [comment]
I can confirm I have the same case:
1) stopped my Vitrual Box
2) installed Docker Desktop
3) did fin project create and created Drupal 8 test project
4) tried to open http://test.docksal but it did not work
5) added both lines in ~/.docksal/docksal.env
6) did fin system reset vhost-proxy
now http://test.docksal and http://test.docksal:8080 work
with or without both lines in ~/.docksal/docksal.env ping test.docksal resolves to 192.168.64.100 and has no issues to ping
fin docker ps looks like this:
`CONTAINER ID        IMAGE                     COMMAND                  CREATED              STATUS                        PORTS                                                    NAMES
0ae06533f44b        docksal/vhost-proxy:1.5   "docker-entrypoint.s…"   About a minute ago   Up About a minute (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
024ccfd916e2        docksal/apache:2.4-2.3    "httpd-foreground"       20 minutes ago       Up 2 minutes (healthy)        80/tcp, 443/tcp                                          test_web_1
41b0b2fed938        docksal/mysql:5.6-1.4     "docker-entrypoint.s…"   20 minutes ago       Up 2 minutes (healthy)        0.0.0.0:32771->3306/tcp                                  test_db_1
d8471282b41a        docksal/cli:2.6-php7.2    "/opt/startup.sh sup…"   20 minutes ago       Up 2 minutes (healthy)        22/tcp, 3000/tcp, 9000/tcp                               test_cli_1
b9a1c102e039        docksal/ssh-agent:1.2     "docker-entrypoint.s…"   39 minutes ago       Up 39 minutes (healthy)                                                                docksal-ssh-agent
088520c35164        docksal/dns:1.1           "docker-entrypoint.s…"   39 minutes ago       Up 39 minutes (healthy)       192.168.64.100:53->53/udp                                docksal-dns
`

**Comment by jessehs** [comment]
I'll report that my situation is still the same. Docksal has been working very well for me with this workaround. One interesting note, however, is that the "https://test.docksal" does not work. It says "connection refused". No matter what I've tried, I can't figure out how to make the SSL connection work. That aspect has been very painful for me since I cannot reproduce a secure SSL environment.

**Comment by sidddrupal** [comment]
I am facing same issue with VPN. outside VPN http://drupaldev.docksal is working fine and with VPN it shows "could not be resolved by DNS".
I added 2 lines to docksal.env file and restarted but no success. 
DOCKSAL_VHOST_PROXY_PORT_HTTP="8080"
DOCKSAL_VHOST_PROXY_IP="0.0.0.0"

Same error for http://drupaldev.docksal:8080 also.

**Comment by lmakarov** [comment]
> "https://test.docksal" does not work

@jessehs I still cannot make sense of your situation... You can override the vhost-proxy HTTPS port 
like this:

```
fin config set --global DOCKSAL_VHOST_PROXY_PORT_HTTPS=8443
fin system reset vhost-proxy
```

In `fin docker ps` output you will see the line for vhost-proxy saying:

```
... 0.0.0.0:8443->443/tcp, 0.0.0.0:8080->80/tcp   docksal-vhost-proxy
```

You should then be able to load `https://test.docksal:8443`

**Comment by lmakarov** [comment]
> I can confirm I have the same case
> ...
> fin docker ps looks like this:
> ```
> ... 192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy
> ```

@vflirt the output above says that neither of the settings where applied.

Here's what the effect would be (starting from a clean Docksal installation):

```
# Default
192.168.64.100:443->443/tcp, 192.168.64.100:80->80/tcp   docksal-vhost-proxy
```

```
# Override vhost-proxy HTTP port
DOCKSAL_VHOST_PROXY_PORT_HTTP=8080

192.168.64.100:443->443/tcp, 192.168.64.100:8080->80/tcp   docksal-vhost-proxy
```

```
# Override vhost-proxy HTTPS port
DOCKSAL_VHOST_PROXY_PORT_HTTPS=8443

192.168.64.100:8443->443/tcp, 192.168.64.100:80->80/tcp   docksal-vhost-proxy
```

```
# Tell vhost-proxy to bind to all network interfaces:
DOCKSAL_VHOST_PROXY_IP="0.0.0.0"

0.0.0.0:443->443/tcp, 0.0.0.0:80->80/tcp   docksal-vhost-proxy
```

**Comment by lmakarov** [comment]
> I am facing same issue with VPN

@sidddrupal this issue is not related to VPN. I put together a new issue with a solution you can try to workaround a greedy VPN connection - https://github.com/docksal/docksal/issues/1124

**Comment by vflirt** [comment]
@jessehs In order to get https working you need to add:
`DOCKSAL_VHOST_PROXY_PORT_HTTPS="8443"`
otherwise https connection will get refused.

@lmakarov probably I pasted old text, here is my current one:
`dobromirhristakiev@~/docksal/expertelite-local$ fin docker ps
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                    PORTS                                         NAMES
131c77113249        docksal/ssh-agent:1.2     "docker-entrypoint.s…"   21 seconds ago      Up 20 seconds (healthy)                                                 docksal-ssh-agent
e643c7c3a3bd        docksal/dns:1.1           "docker-entrypoint.s…"   29 seconds ago      Up 28 seconds (healthy)   192.168.64.100:53->53/udp                     docksal-dns
f8f4eda056f8        docksal/vhost-proxy:1.5   "docker-entrypoint.s…"   31 seconds ago      Up 30 seconds (healthy)   0.0.0.0:8080->80/tcp, 0.0.0.0:8443->443/tcp   docksal-vhost-proxy
8a752176ffc4        docksal/apache:2.4-2.3    "httpd-foreground"       12 days ago         Up 11 minutes (healthy)   80/tcp, 443/tcp                               expertelite-local_web_1
987982cde32b        docksal/cli:2.6-php7.2    "/opt/startup.sh sup…"   12 days ago         Up 11 minutes (healthy)   22/tcp, 3000/tcp, 9000/tcp                    expertelite-local_cli_1
372c1704f4fe        docksal/mysql:5.6-1.4     "docker-entrypoint.s…"   4 weeks ago         Up 11 minutes (healthy)   0.0.0.0:32768->3306/tcp                       expertelite-local_db_1
`

**Comment by achekulaev** [comment]
Closing for inactivity
